src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -72,6 +78,13 @@ public sealed class RestoreManagerPackage : AsyncPackage
                 _buildEvents = dte.Events.BuildEvents;
                 _buildEvents.OnBuildBegin += BuildEvents_OnBuildBegin;
 
+                var solutionBuildManager = (IVsSolutionBuildManager5)await GetServiceAsync(typeof(SVsSolutionBuildManager));
+                if (solutionBuildManager != null)
+                {
+                    uint updateSolutionEventsCookie4;
```

Comment:
Please do not throw away the cookie. Store it in a field and unadvise in your package close method.